tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex

```diff
@@ -14,13 +14,15 @@
 
 \tikzset{decorate/.is if=tikz@decoratepath,
   /pgf/decoration/name/.store in=\tikz@decoration@name,
-  /pgf/decoration/.unknown/.code=%
-    \pgfifdecoration{\pgfkeyscurrentname}{\edef\tikz@decoration@name{\pgfkeyscurrentname}}
-    {\pgfifmetadecoration{\pgfkeyscurrentname}{\edef\tikz@decoration@name{\pgfkeyscurrentname}}
-    % Fully expand `\pgfkeyscurrentname' before being used in first-arg of
-    % `/errors/unknown key'.
-    {\pgfkeys{/errors/unknown
-        key/.expanded={/pgf/decoration/\pgfkeyscurrentname}{\pgfutil@unexpanded{#1}}}}},%
+  /pgf/decoration/.unknown/.code={%
+    \pgfifdecoration{\pgfkeyscurrentname}
+      {\edef\tikz@decoration@name{\pgfkeyscurrentname}}
+      {\pgfifmetadecoration{\pgfkeyscurrentname}
+        {\edef\tikz@decoration@name{\pgfkeyscurrentname}}
+        {\tikzerror{Unknown decoration option '\pgfkeyscurrentname'.
+            I do not know the key '/pgf/decoration/\pgfkeyscurrentname'
+            and I am going to ignore it. Perhaps you misspelled it}}}%
```

Comment:
Shouldn't we somehow keep the `/errors/unknown key` handler?

Reply:
Using `/error/unknown key` handler, the full error message looks like
```
! Package pgfkeys Error: I do not know the key '/pgf/decoration/red' and 
I am going to ignore it. Perhaps you misspelled it.

See the pgfkeys package documentation for explanation.
```
which seems less helpful as user should really be led to the tikz doc, not pgfkeys doc.

But I confess current `\tikzerror` way is not good too.

Reply:
Then maybe an additional error handler for decorations?

Reply:
I decided to save time and just switch back to `/errors/unknown key`.